### DISCUSSION TITLE
Content-Type `application/octet-stream` set By Default

### DISCUSSION THREAD

**Discussion by efiege** [eclipse-edc/Connector#3783] [General]
Dear EDC-Team,

we have made the observation, that the `Content-Type`-header of the request, that is send to the consumer, gets automatically set to `application/octet-stream` by default. The observation was made, when executing a `ProviderPush`. 

Is this the intended behaviour?

We would expect that the `Content-Type`-header gets forwarded from the producer over the data-plane to the consumer. Such that this information does not get lost.

@SebastianOpriel

**Comment by ndr-brt** [comment]
Theoretically the `Content-Type` is extracted from the incoming request and set into the `DataFlowRequest.properties` field:
https://github.com/eclipse-edc/Connector/blob/a1957129160dde68850b52b592e6e7404eeb7199/extensions/data-plane/data-plane-public-api/src/main/java/org/eclipse/edc/connector/dataplane/api/controller/DataFlowRequestSupplier.java#L65-L70

Then it is set on the `contentType` of the `HttpRequestParam` object:
https://github.com/eclipse-edc/Connector/blob/a1957129160dde68850b52b592e6e7404eeb7199/extensions/data-plane/data-plane-http/src/main/java/org/eclipse/edc/connector/dataplane/http/params/decorators/BaseSourceHttpParamsDecorator.java#L46-L51

that it's used to create the `RequestBody` object with the `contentType`:
https://github.com/eclipse-edc/Connector/blob/a1957129160dde68850b52b592e6e7404eeb7199/extensions/data-plane/data-plane-http/src/main/java/org/eclipse/edc/connector/dataplane/http/params/HttpRequestFactory.java#L74-L82

so, theoretically it should work, if that's not the case, I'd say that it could be a bug.

**Reply by efiege** [reply]
Thanks for this fast answer. 

The `DataFlowRequestSupplier` you mentioned is only used from the `DataPlanePublicApiController` if I see it correctly. As a result it is used for the `ConsumerPull` and not used for the `ProviderPush` method. The observation we made was related to the `ProviderPush`.

You also mentioned that 
> the `Content-Type` is extracted from the incoming request

but our observation was related to passing the `Content-Type` header, which is coming from the response of the provider-system, through the `data-plane` into the consumer-system.

If this is still not intended, I will create a bug report,

**Reply by ndr-brt** [reply]
whops, I overlooked that detail, yes, looking at the implementation seems that no `Content-Type` is extracted from the response and by consequence, it is not forwarded to the consumer. 
It definitely deserves an issue.
Ref.: 
https://github.com/eclipse-edc/Connector/blob/e6522fe632d01765a0b51f0a193b1ac9b52cfcd2/extensions/data-plane/data-plane-http/src/main/java/org/eclipse/edc/connector/dataplane/http/pipeline/HttpDataSource.java#L59
